ecl/hql/hqlutil.cpp

```diff
@@ -9854,9 +9854,13 @@ static IFieldFilter * createIfBlockFilter(IRtlFieldTypeDeserializer &deserialize
 unsigned buildRtlRecordFields(IRtlFieldTypeDeserializer &deserializer, unsigned &idx, const RtlFieldInfo * * fieldsArray, IHqlExpression *record, IHqlExpression *rowRecord)
 {
     unsigned typeFlags = 0;
+    unsigned numFields = getFlatFieldCount(record);
+    unsigned payloadFields = numPayloadFields(record);
     ForEachChild(i, record)
     {
         unsigned fieldFlags = 0;
+        if (idx >= numFields-payloadFields)
```

Comment:
I'm not sure that this idx test is correct if it is derived from getFlatFieldCount - if the index record happened to extend another index.  (very weird code).  firstPayloadField() may be safer.